nnvm/python/nnvm/frontend/keras.py

```diff
@@ -486,14 +486,29 @@ def from_keras(model):
                 raise TypeError("Unknown layer type or unsupported Keras version : {}"
                                 .format(keras_layer))
             for node in inbound_nodes:
-                for pred in node.inbound_layers:
-                    predecessors.append(pred.name)
-            if len(predecessors) == 1:
-                insym = symtab.get_var(predecessors[0], must_contain=True)
-            else:
+                # Since Keras allows creating multiple layers from the same name instance,
+                # we append node index to the symbol name to make it unique.
+                # The one exception is InputLayer.  Changing input variable names after conversion
+                # would confuse users, so we should keep them as far as possible.  Fortunately,
+                # it's safe because all the input layers have unique names in Keras.
+                for i, pred in zip(node.node_indices, node.inbound_layers):
+                    if isinstance(pred, keras.engine.topology.InputLayer):
+                        predecessors.append(pred.name)
+                    else:
+                        predecessors.append(pred.name + str(i))
+
+            # With the current implementation, _convert_merge and _convert_concat
+            # expects their inputs to be List.  For now, we handle them differently,
+            # but in future, we could have a unified way here.
```

Comment:
@kazum  Keras can't create multiple merge or concat layers from the same name instance, is this correct? Seems this is one assumption here.

Another assumption is keras layers other than merge or concat have only one predecessor. I think this is correct. Can you verify it?

Reply:
> @kazum Keras can't create multiple merge or concat layers from the same name instance, is this correct? Seems this is one assumption here.

I've confirmed it's not correct.  I'll add a test case for that and fix the problem.

> Another assumption is keras layers other than merge or concat have only one predecessor. I think this is correct. Can you verify it?

Fortunately, looks like the assumption is no longer necessary in the fixed version.